modules/services/kanshi.nix

```diff
@@ -0,0 +1,201 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.kanshi;
+
+  outputModule = types.submodule {
+    options = {
+
+      criteria = mkOption {
+        type = types.str;
+        description = ''
+          <literallayout>
+          The criteria can either be an output name, an output description or "*".
+          The latter can be used to match any output.
+
+          On sway(1), output names and descriptions can be obtained via
+          <literal>swaymsg -t get_outputs</literal>.
+          </literallayout>
+        '';
+      };
+
+      status = mkOption {
+        type = types.nullOr (types.enum [ "enable" "disable" ]);
+        default = null;
+        description = ''
+          Enables or disables the specified output.
+        '';
+      };
+
+      mode = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "1920x1080@60Hz";
+        description = ''
+          <literallayout>
+          &lt;width&gt;x&lt;height&gt;[@&lt;rate&gt;[Hz]]
+
+          Configures the specified output to use the specified mode.
+          Modes are a combination of width and height (in pixels) and
+          a refresh rate (in Hz) that your display can be configured to use.
+          </literallayout>
+        '';
+      };
+
+      position = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "1600,0";
+        description = ''
+          <literallayout>
+          &lt;x&gt;,&lt;y&gt;
+
+          Places the output at the specified position in the global coordinates
+          space.
+          </literallayout>
+        '';
+      };
+
+      scale = mkOption {
+        type = types.nullOr types.str;
```

Comment:
Perhaps `nullOr float` would be more appropriate? Seems like that's what the parser expects: https://github.com/emersion/kanshi/blob/846b725637840fd41aaa252d457392758197e1d8/parser.c#L323

Reply:
I have updated it. The only thing that I do not like too much is that `toString` outputs a float with a fixed precision. Maybe I should do like in https://github.com/NixOS/nixpkgs/blob/5f5caaceb4b6f886e608c11f4a0ffb828e8b0ca0/nixos/modules/services/x11/picom.nix#L15 ?